tests/Utils/Test_ShardSizeCalculator.cpp

```diff
@@ -17,131 +17,64 @@
-}
-
-BOOST_AUTO_TEST_CASE(test_one_shard_normal) {
-  INIT_STDOUT_LOGGER();
-  uint32_t result = ShardSizeCalculator::CalculateShardSize(700);
-  BOOST_CHECK_MESSAGE(result == 651,
-                      "Expected: 651. Result: " + to_string(result));
-}
-
-BOOST_AUTO_TEST_CASE(test_one_shard_lower_bound) {
-  INIT_STDOUT_LOGGER();
-  uint32_t result = ShardSizeCalculator::CalculateShardSize(651);
-  BOOST_CHECK_MESSAGE(result == 651,
-                      "Expected: 651. Result: " + to_string(result));
-}
-
-BOOST_AUTO_TEST_CASE(test_one_shard_upper_bound) {
-  INIT_STDOUT_LOGGER();
-  uint32_t result = ShardSizeCalculator::CalculateShardSize(1367);
-  BOOST_CHECK_MESSAGE(result == 651,
-                      "Expected: 651. Result: " + to_string(result));
-}
-
-BOOST_AUTO_TEST_CASE(test_two_shards_lower_bound) {
-  INIT_STDOUT_LOGGER();
-  uint32_t result = ShardSizeCalculator::CalculateShardSize(1368);
-  BOOST_CHECK_MESSAGE(result == 684,
-                      "Expected: 684. Result: " + to_string(result));
-}
-
-BOOST_AUTO_TEST_CASE(test_two_shards_upper_bound) {
-  INIT_STDOUT_LOGGER();
-  uint32_t result = ShardSizeCalculator::CalculateShardSize(2131);
-  BOOST_CHECK_MESSAGE(result == 684,
-                      "Expected: 684. Result: " + to_string(result));
-}
-
-BOOST_AUTO_TEST_CASE(test_ten_shards) {
-  INIT_STDOUT_LOGGER();
-  uint32_t result = ShardSizeCalculator::CalculateShardSize(8000);
-  BOOST_CHECK_MESSAGE(result == 771,
-                      "Expected: 771. Result: " + to_string(result));
-}
-
-BOOST_AUTO_TEST_CASE(test_ten_shards_lower_bound) {
-  INIT_STDOUT_LOGGER();
-  uint32_t result = ShardSizeCalculator::CalculateShardSize(7710);
-  BOOST_CHECK_MESSAGE(result == 771,
-                      "Expected: 771. Result: " + to_string(result));
-}
-
-BOOST_AUTO_TEST_CASE(test_ten_shards_upper_bound) {
-  INIT_STDOUT_LOGGER();
-  uint32_t result = ShardSizeCalculator::CalculateShardSize(8579);
-  BOOST_CHECK_MESSAGE(result == 771,
-                      "Expected: 771. Result: " + to_string(result));
-}
-
-BOOST_AUTO_TEST_CASE(test_max_shards_normal) {
-  INIT_STDOUT_LOGGER();
-  uint32_t result = ShardSizeCalculator::CalculateShardSize(20000);
-  BOOST_CHECK_MESSAGE(result == 816,
-                      "Expected: 816. Result: " + to_string(result));
-}
-
-BOOST_AUTO_TEST_CASE(test_max_shards_lower_bound) {
-  INIT_STDOUT_LOGGER();
-  uint32_t result = ShardSizeCalculator::CalculateShardSize(19584);
-  BOOST_CHECK_MESSAGE(result == 816,
-                      "Expected: 816. Result: " + to_string(result));
+typedef std::vector<uint32_t> TestDataSet;
+typedef std::map<uint32_t, TestDataSet> ShardSizeMap;
+
+void prepareTestdata(ShardSizeMap& testData) {
+  TestDataSet tds;
+  testData[651] = tds = {0, 1, 650, 651, 1000, 1367};
+  testData[684] = tds = {1368, 2131, 2132};
+  testData[711] = tds = {2133, 2200, 2867};
+  ;
+  testData[717] = tds = {2868, 2869, 3674};
```

Comment:
Once libTestUtils is merged, maybe we can randomize the middle value within the range?

Reply:
By my opinion randomizing the middle value is not necessary since this test focuses on the boundary values, so the middle value just confirms the "equivalence class" where the two boundaries belong. However randomizing the middle value would cause no harm so do as you like.

Reply:
Agree, done.